operator/cmd/mesh/manifest-generate_test.go

```diff
@@ -451,14 +450,19 @@ func findObject(objs object.K8sObjects, name, kind string) *object.K8sObject {
 // -ldflags "-X istio.io/pkg/version.buildHub=myhub -X istio.io/pkg/version.buildVersion=mytag"
 // results in these values showing up in a generated manifest.
 func TestLDFlags(t *testing.T) {
+	testDataDir = filepath.Join(repoRootDir, "cmd/mesh/testdata/manifest-generate")
 	tmpHub, tmpTag := version.DockerInfo.Hub, version.DockerInfo.Tag
 	defer func() {
 		version.DockerInfo.Hub, version.DockerInfo.Tag = tmpHub, tmpTag
 	}()
 	version.DockerInfo.Hub = "testHub"
 	version.DockerInfo.Tag = "testTag"
 	l := NewLogger(true, os.Stdout, os.Stderr)
-	_, iops, err := GenerateConfig(nil, "", true, nil, l)
+	ysf, err := yamlFromSetFlags([]string{"installPackagePath=" + filepath.Join(testDataDir, "data-snapshot")}, false, l)
```

Comment:
why does this use the fake snapshot?

Reply:
this is testing logic not the templates i.e. if the templates change, we don't want this golden file to change right?

Reply:
We are just checking the IOP has the right hub and tag. In what case could a template change and cause the test to fail that is not a legitimate bug in the template?

Reply:
The more tests we are like this, the more likely someone is going to need to update the datasnapshot for their PR, and now all of the sudden there 1 line PR has 4k lines of code changed and 10 tests failing

Reply:
oh, i missed the fact that this is not testing against a golden. you are right, i'll change this.

Reply:
done